From ea57b4998912f439058db9fd5450afdeb94a5d0a Mon Sep 17 00:00:00 2001
From: Ralf Gliese <gliese@elrest.de>
Date: Wed, 18 Nov 2020 12:53:12 +0100
Subject: [PATCH] config: dts: ec: enable PTN5150 driver and adjust ec300
 device tree

activate PTN5150 driver for USB-C interface of EC300 controller
and add configure options for u_ether gadget

Signed-off-by: Ralf Gliese <gliese@elrest.de>
Signed-off-by: Christian Hohnstaedt <Christian.Hohnstaedt@wago.com>
---
 arch/arm/boot/dts/imx6q-ec-752_8303.dts |  9 +++++++++
 arch/arm/boot/dts/imx6q-ec.dtsi         | 16 ++++++++++++++++
 arch/arm/configs/imx6_vtpctp_defconfig  | 13 +++++++++++--
 3 files changed, 36 insertions(+), 2 deletions(-)

diff --git a/arch/arm/boot/dts/imx6q-ec-752_8303.dts b/arch/arm/boot/dts/imx6q-ec-752_8303.dts
index 7ca52fa..10dbfbd 100644
--- a/arch/arm/boot/dts/imx6q-ec-752_8303.dts
+++ b/arch/arm/boot/dts/imx6q-ec-752_8303.dts
@@ -20,3 +20,12 @@
 	pinctrl-0 = <&pinctrl_hdmi>;
 	status = "okay";
 };
+
+&usbotg {
+	dr_mode = "otg";
+	status = "okay";
+};
+
+&cc_logic_usb_c {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6q-ec.dtsi b/arch/arm/boot/dts/imx6q-ec.dtsi
index 3651e6b..3d9860c 100644
--- a/arch/arm/boot/dts/imx6q-ec.dtsi
+++ b/arch/arm/boot/dts/imx6q-ec.dtsi
@@ -483,6 +483,15 @@
 		};
 	};
 
+	cc_logic_usb_c: cclogic@1e {
+		compatible = "nxp,ptn5150";
+		reg = <0x1e>;
+		int-gpio = <&gpio1 3 GPIO_ACTIVE_HIGH>;
+		vbus-gpio = <&gpio1 24 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&ptn5150_pins>;
+	};
+
 };
 
 &i2c3 {
@@ -766,6 +775,13 @@
 			>;
 		};
 
+		ptn5150_pins: pinmux_ptn5150_pins {
+			fsl,pins = <
+				/* IRQ USB-C */
+				MX6QDL_PAD_GPIO_3__GPIO1_IO03	0x1b0b0
+			>;
+		};
+
 	};
 };
 
diff --git a/arch/arm/configs/imx6_vtpctp_defconfig b/arch/arm/configs/imx6_vtpctp_defconfig
index 21c3b0a..47c775e 100644
--- a/arch/arm/configs/imx6_vtpctp_defconfig
+++ b/arch/arm/configs/imx6_vtpctp_defconfig
@@ -58,8 +58,6 @@ CONFIG_MAC_PARTITION=y
 CONFIG_CFQ_GROUP_IOSCHED=y
 CONFIG_ARCH_MXC=y
 CONFIG_SOC_IMX6Q=y
-CONFIG_IMX6Q_USE_EXTERNAL_ENET_CLK=y
-CONFIG_IMX6Q_SET_IMX6QDL_CLK_CKO2_12MHZ=y
 CONFIG_SOC_IMX6SL=y
 CONFIG_SOC_IMX6SX=y
 CONFIG_WAGO_SYSTEM_BASED_STARTUP=y
@@ -479,6 +477,16 @@ CONFIG_USB_MXS_PHY=y
 CONFIG_USB_ULPI=y
 CONFIG_USB_GADGET=y
 CONFIG_USB_FSL_USB2=y
+CONFIG_USB_CONFIGFS=m
+CONFIG_USB_CONFIGFS_ACM=y
+CONFIG_USB_CONFIGFS_NCM=y
+CONFIG_USB_CONFIGFS_ECM=y
+CONFIG_USB_CONFIGFS_ECM_SUBSET=y
+CONFIG_USB_CONFIGFS_RNDIS=y
+CONFIG_USB_CONFIGFS_EEM=y
+CONFIG_USB_CONFIGFS_MASS_STORAGE=y
+CONFIG_USB_CONFIGFS_F_FS=y
+CONFIG_USB_ETH=m
 CONFIG_USB_G_SERIAL=m
 CONFIG_MMC=y
 CONFIG_MMC_SDHCI=y
@@ -508,6 +516,7 @@ CONFIG_UIO_DMEM_GENIRQ=y
 CONFIG_UIO_PRUSS=y
 CONFIG_STAGING=y
 CONFIG_STAGING_MEDIA=y
+CONFIG_EXTCON_PTN5150=m
 CONFIG_IIO=y
 CONFIG_VF610_ADC=y
 CONFIG_PWM=y
-- 
2.7.4

